src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -403,6 +407,31 @@ public static Vector256<ushort> Load256(ref byte ptr)
                 (Vector128<ushort> lower, Vector128<ushort> upper) = Vector128.Widen(Vector128.LoadUnsafe(ref ptr));
                 return Vector256.Create(lower, upper);
             }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            [CompExactlyDependsOn(typeof(Avx))]
+            public static bool Compare256(ref byte left, ref ushort right)
```

Comment:
It is not clear what this function does. Is `left` `utf8` while `right` is `utf16` ? If so - why do you widen the utf8 when you can narrow the utf16 assuming they're ASCII ?

Reply:
> Is left utf8 while right is utf16 ?

Yes.

> why do you widen the utf8 when you can narrow the utf16 assuming they're ASCII ?

We are assuming they are ascii, but don't know they are. If we narrow the utf16 we'll lose the high bits and incorrectly compare utf8 with truncated utf16. So [DINGBAT NEGATIVE CIRCLED DIGIT ONE](https://www.fileformat.info/info/unicode/char/2776/index.htm) would incorrectly match with 'v'.

> It is not clear what this function does

Naming suggestions are welcome! Although note, with the current `TLoader` pattern this is a generic method used for both byte-byte (utf8-utf8) comparison and byte-ushort (utf8-utf16) comparison.